src/server/server-quiz/clients/trivia-client.js

```diff
@@ -0,0 +1,36 @@
+const { default: axios } = require("axios");
+
+async function fetchQuestions(url) {
+  const errorResponse = {
+    error: true,
+    description: `Error fetching questions`,
+  };
+  try {
+    const response = await axios.get(url).then((response) => response);
+    const questions = response.data.results;
+
+    return {
+      error: false,
+      questions: questions,
+    };
+  } catch (error) {
+    return errorResponse;
+  }
+}
+
+async function fetchMultipleTopics(urlArray) {
+  let allQuestions = [];
+
+  for (let url of urlArray.urlArray) {
+    let topicQuestions = await fetchQuestions(url);
+
+    allQuestions = [...allQuestions, ...topicQuestions.questions];
```

Comment:
Why do you reassign the array and not just push a new element? I know in some cases it is important, but do not know, why is it important now

Reply:
The reason is that each topic has it's own questions so you get something like:
{
questions[...] 
},
{
questions{...}
}
so I wanted to spread all to one array